### PR TITLE
AC Automated Test - LRAC-9867 | master

### PR DESCRIPTION
**Pull request by ViniLopes20**
https://issues.liferay.com/browse/LRAC-9867 (Automation ticket)
https://issues.liferay.com/browse/LRAC-7832 (Bug 1)
https://issues.liferay.com/browse/LRAC-11043 (Bug 2)
[Testray Result](https://testray.liferay.com/home/-/testray/case_results?testrayBuildId=1879500244&testrayRunId=1879502784)

### CODE REVIEW

**Comment by ViniLopes20** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/FormEvents.testcase [hunk 1]
```diff
@@ -65,6 +65,175 @@ definition {
...
+				}
+
+				task ("Click to focus on rich text writing field") {
+					if ("${fieldName}" == "Rich Text") {
+						FormFields.addRichTextContent(
+							content = "",
+							fieldLabel = "Rich Text");
+					}
+				}
+
```
Comment (manuferreira) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/FormEvents.testcase:119]: Hey @ViniLopes20 can you tell me if I'm understanding this block of code right?
So we have an `if`just for `Date and Time` :ok: 
We have an `if`just when the fields are different than `Rich Text` and `Date and Time`, so this leaves us to check for: `Date,Image,Upload,Color` :ok: 
and the last `if` checks for `Rich Text` :ok: 
I found it a little bit confusing, do you think using `else if` statements (like [this example here](https://github.com/liferay/liferay-portal/blob/master/portal-web/test/functional/com/liferay/portalweb/tests/enduser/documentmanagement/documentsadministration/DMAdmin.testcase#L241#L255)) would be better? I know the code will get a little bit bigger, but I want to know your opinion on it.

Comment (ViniLopes20) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/FormEvents.testcase:119]: Hey @manuferreira, I remember Alex had told me that he thinks using **else if** or **else** is a little bad for code legibility. He gave me this [video here](https://www.youtube.com/watch?v=EumXak7TyQ0). I think our situation is different from the case of the video that will use a return to not have to do all the conditions. So I think we could use **else if** here to save some of the Poshi resources. But I think in relation to which one is easier to understand, I think using the **if** makes it a little easier to read the test logic.

Comment (ViniLopes20) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/FormEvents.testcase:119]: We could also use the **else** in this case, but I think that the use of the **else** will complicate the reading of the code that a person could have, the use of the **else** does not make it so clear which cases will be included in it, so I think that we shouldn't use in this situation to be clear what the fields are.

File: portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uielements/Button.path [hunk 2]
```diff
@@ -255,6 +255,11 @@
 	<td>//button[*[name()='svg'][contains(@class,'lexicon-icon-effects')]]</td>
 	<td></td>
 </tr>
+<tr>
+	<td>COLOR_PICKER_BUTTON</td>
+	<td>//div[contains(@class,'input-group')]//button[contains(@class,'clay-color-btn')]</td>
+	<td></td>
+</tr>
 <tr>
 	<td>COLOR_PICKER</td>
```
Comment (christianstokes) [inline portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uielements/Button.path:264]: Not related to your code, @ViniLopes20, but WOW look at this path haha 😅

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/FormEvents.testcase [hunk 3]
```diff
@@ -65,6 +65,171 @@ definition {
...
+							fieldLabel = "Rich Text");
+					}
+					else if (("${fieldName}" != "Rich Text") && ("${fieldName}" != "Date and Time")) {
+						Click(
+							key_fieldName = "${fieldName}",
+							locator1 = "FormFields#GENERIC_INPUT_FIELD");
+					}
+				}
+
+				task ("Wait for sending requests") {
```
Comment (christianstokes) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/FormEvents.testcase:116]: This pause and the other ones are fine for now but later we should refactor this into a macro

**Review by christianstokes — APPROVED** [review]
Really nice test @ViniLopes20 great work. Approved.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/FormEvents.testcase (hunk 3 above)
Comment (ViniLopes20) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/FormEvents.testcase:116]: Yes, I left the pauses for us to investigate and resolve them in the ticket you opened, I think this makes it easier for us to organize and resolved. Thanks, Chris.

**Comment by christianstokes** [comment]
ci:forward

**Pull request closed without merging**